File: 17-Functions-Part-1/42-Default-Parameters.py
# === first way ===
def speak(animal='dog'):
    if animal == 'dog':
        return 'woof'
    elif animal == 'pig':
        return 'oink'
    elif animal == 'duck':
        return 'quack'
    elif animal == 'cat':
        return 'meow'
    else:
        return '?'

# === second way ===
def speak(animal='dog'):
    noises = {
        'dog': 'woof',
        'pig': 'oink',
        'duck': 'quack',
        'cat': 'meow',
    }
    return noises.get(animal, '?')  # returns '?' if no animal is found

File: 17-Functions-Part-1/42-Default-Parameters.js
// === first way ===
function speak(animal='dog') {
    let noise
    switch (animal) {
        case 'dog':
            noise = 'woof'
            break
        case 'pig':
            noise = 'oink'
            break
        case 'duck':
            noise = 'oink'
            break
        case 'cat':
            noise = 'meow'
            break
        default:
            noise = '?'
    }
    return noise
}

// === second way ===
function speak2(animal='dog') {
    const noises = {
        dog: 'woof',
        pig: 'oink',
        duck: 'quack',
        cat: 'moew'
    }
    return noises[animal] || '?'
}
